pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -625,6 +625,16 @@ public void start() throws Exception {
         this.updateBrokerDispatchThrottlingMaxRate();
         this.startCheckReplicationPolicies();
         this.startDeduplicationSnapshotMonitor();
+        this.startClearInvalidateTopicNameCacheTask();
+    }
+
+    protected void startClearInvalidateTopicNameCacheTask() {
+        final int maxSecondsToClearTopicNameCache = pulsar.getConfiguration().getMaxSecondsToClearTopicNameCache();
+        inactivityMonitor.scheduleAtFixedRate(
+            () -> TopicName.clearIfReachedMaxCapacity(pulsar.getConfiguration().getTopicNameCacheCaxCapacity()),
+            maxSecondsToClearTopicNameCache,
+            maxSecondsToClearTopicNameCache,
+            TimeUnit.SECONDS);
```

Comment:
This solution doesn't seem effective when the cache size can go over limits. I think it's better to use Caffeine cache with max size (as a LRU cache). You could remove expiration when using Caffeine if that's expensive.

Reply:
> This solution doesn't seem effective when the cache size can go over limits. I think it's better to use Caffeine cache with max size (as a LRU cache). You could remove expiration when using Caffeine if that's expensive.

- Clearing cache is just for the topic that was deleted, if it was not deleted, leaving it in the memory is better.
- Once there are too many topics in use, Caffeine will trigger queue items switching, which will cause more CPU costs.